File: src-server/modules/content/api/friend_link.go
package api

import (
	"github.com/zhouhp1295/g3-cms/boot"
	"github.com/zhouhp1295/g3-cms/modules/content/dao"
	"github.com/zhouhp1295/g3/net"
)

type _contentFriendLinkApi struct {
	net.BaseApi
}

var ContentFriendLinkApi = &_contentFriendLinkApi{
	net.BaseApi{Dao: dao.ContentFriendLinkDao},
}

const (
	PermContentFriendLinkList   = "content:friendLink:list"
	PermContentFriendLinkQuery  = "content:friendLink:query"
	PermContentFriendLinkAdd    = "content:friendLink:add"
	PermContentFriendLinkEdit   = "content:friendLink:edit"
	PermContentFriendLinkRemove = "content:friendLink:remove"
)

func init() {
	//基础接口
	boot.ApiGet("/admin/content/friendLink/page", ContentFriendLinkApi.HandlePage, PermContentFriendLinkQuery)
	boot.ApiGet("/admin/content/friendLink/get", ContentFriendLinkApi.HandleGet, PermContentFriendLinkQuery)
	boot.ApiPost("/admin/content/friendLink/insert", ContentFriendLinkApi.HandleInsert, PermContentFriendLinkAdd)
	boot.ApiPut("/admin/content/friendLink/update", ContentFriendLinkApi.HandleUpdate, PermContentFriendLinkEdit)
	boot.ApiPut("/admin/content/friendLink/status", ContentFriendLinkApi.HandleUpdateStatus, PermContentFriendLinkEdit)
	boot.ApiDelete("/admin/content/friendLink/delete", ContentFriendLinkApi.HandleDelete, PermContentFriendLinkRemove)
}

File: src-backend/src/api/content/friend_link.js
import request from '@/utils/request'

// 查询标签列表
export function listFriendLink(query) {
  return request({
    url: '/api/admin/content/friendLink/page',
    method: 'get',
    params: query
  })
}

// 查询标签详细
export function getFriendLink(id) {
  return request({
    url: '/api/admin/content/friendLink/get',
    method: 'get',
    params: {
      id
    }
  })
}

// 新增标签配置
export function addFriendLink(data) {
  return request({
    url: '/api/admin/content/friendLink/insert',
    method: 'post',
    data: data
  })
}

// 修改标签配置
export function updateFriendLink(data) {
  return request({
    url: '/api/admin/content/friendLink/update',
    method: 'put',
    data: data
  })
}

// 删除标签配置
export function delFriendLink(id) {
  return request({
    url: '/api/admin/content/friendLink/delete',
    method: 'delete',
    params: {
      id
    }
  })
}

// 推荐状态修改
export function changeFriendLinkStatus(id, status) {
  const data = {
    id,
    status
  }
  return request({
    url: '/api/admin/content/friendLink/status',
    method: 'put',
    data: data
  })
}
